cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -3758,40 +3759,48 @@ public void setSubscriptionChanged(boolean subscriptionChanged) {
    * If get consent 1 in IABTCF_VendorConsents at position 1139 then perform subscribe or tracking according to IabTcfMode
    */
   private void setTCF() {
-    if (getIabTcfMode() == IabTcfMode.TRACKING_WAIT_FOR_CONSENT) {
-      setTrackingConsentRequired(true);
-    }
-    if (getIabTcfMode() == IabTcfMode.SUBSCRIBE_WAIT_FOR_CONSENT) {
-      setSubscribeConsentRequired(true);
-    }
-    Context mContext = context.getApplicationContext();
-    SharedPreferences mPreferences = PreferenceManager.getDefaultSharedPreferences(mContext);
-
-    SharedPreferences.OnSharedPreferenceChangeListener mListener;
-    mListener = (preferences, key) -> {
-      if (key.equals(IABTCF_VendorConsents)) {
-        String vendorConsents = mPreferences.getString(IABTCF_VendorConsents, "0");
-        if (vendorConsents.length() > IABTCF_VendorConsent_POSITION - 1) {
-          char consentStatus = vendorConsents.charAt(IABTCF_VendorConsent_POSITION - 1); // charAt uses zero-based indexing, so the 1139th character is at index 1138.
-          boolean hasConsent = (consentStatus == '1');
-
-          if (hasConsent) {
-            if (getIabTcfMode() == IabTcfMode.TRACKING_WAIT_FOR_CONSENT) {
-              setTrackingConsent(true);
-            }
-            if (getIabTcfMode() == IabTcfMode.SUBSCRIBE_WAIT_FOR_CONSENT) {
-              setSubscribeConsent(true);
+    try {
+      if (getIabTcfMode() == IabTcfMode.TRACKING_WAIT_FOR_CONSENT) {
```

Comment:
This could be written more concisely:
```
IabTcfMode mode = getIabTcfMode();
setTrackingConsentRequired(mode == IabTcfMode.TRACKING_WAIT_FOR_CONSENT); 
setSubscribeConsentRequired(mode == IabTcfMode.SUBSCRIBE_WAIT_FOR_CONSENT);
```